taichi/codegen/spirv/spirv_codegen.cpp

```diff
@@ -516,13 +516,18 @@ class TaskCodegen : public IRVisitor {
   }
 
   void visit(ReturnStmt *stmt) override {
+    // Now we only support one ret
+    const auto &ret_attribs = ctx_attribs_->rets()[0];
+    auto dt = ret_attribs.dt;
+    if (auto tensor_type = dt->cast<TensorType>()) {
+      dt = tensor_type->get_element_type();
+    }
     for (int i = 0; i < stmt->values.size(); i++) {
-      const auto &ret_attribs = ctx_attribs_->rets()[i];
-      const auto dt = ret_attribs.dt;
       spirv::Value buffer_val = ir_->make_value(
           spv::OpAccessChain,
           ir_->get_storage_pointer_type(ir_->get_primitive_type(dt)),
-          get_buffer_value(BufferType::Rets, PrimitiveType::i32),
+          get_buffer_value(BufferType::Rets, dt),
```

Comment:
Sorry I don't quite get it here. Why would this be able to return a matrix?

Reply:
Now we only support return one value, so if the return type is the matrix, its elements will be involved in `stmt->values`. Likewise, `ctx_attribs_->rets()`'s size will always be one. `dt` is the primitive type of the matrix element. And in the spir-v backend, I use a StructType to include an ArrayType, the matrix will be squeezed to an array when compiling. So to access the element, I change the argument of `OpAccessChain`, `0,i` instead of `i`.

Reply:
For the scalar return type, I also use an ArrayType to involve this return value. So I can do the same operations when dealing with matrix and scalar.